backend/main.py

```
@@ -32,37 +30,70 @@ class DevRAIApplication:
         except Exception as e:
             logger.error(f"Error during background task startup: {e}", exc_info=True)
             await self.stop_background_tasks()
             raise
 
     async def test_weaviate_connection(self):
         """Test Weaviate connection during startup."""
+        if not settings.code_intelligence_enabled:
+            logger.info("Weaviate Is for Full Mode")
+            return 
         try:
             async with get_weaviate_client() as client:
```

Ruff (0.14.14)

Undefined name `get_weaviate_client`

(F821)